components/mpas-ocean/src/shared/mpas_ocn_diagnostics.F

```diff
@@ -1627,30 +1633,116 @@ subroutine ocn_diagnostic_solve_vortVel(relativeVorticity, &
       !
       ! Compute divergence, kinetic energy, and vertical velocity
       !
-      allocate(div_hu(nVertLevels),div_huTransport(nVertLevels),div_huGMBolus(nVertLevels))
+      allocate(div_hu(nVertLevels),div_huTransport(nVertLevels))
+
+      if (config_submesoscale_enable) then
+         allocate(div_huMLEBolus(nVertLevels))
+#ifdef MPAS_OPENACC
+         !$acc parallel loop gang vector &
+         !$acc          present(nEdgesOnCell, edgesOnCell, edgeSignonCell, dcEdge, dvEdge, &
+         !$acc                  areaCell, layerThicknessEdgeFlux, normalMLEVelocity, &
+         !$acc                  vertMLEBolusVelocityTop, invAreaCell, minLevelCell) &
+         !$acc          private(invAreaCell1, i, iEdge, edgeSignOnCell_temp, dcEdge_temp, &
+         !$acc                  dvEdge_temp, k)
+#else
+         !$omp parallel
+         !$omp do schedule(runtime) &
+         !$omp private(i, iEdge, invAreaCell1, edgeSignOnCell_temp, dcEdge_temp, dvEdge_temp, &
+         !$omp         k, div_huMLEBolus)
+#endif
+         do iCell = 1, nCells
+            div_huMLEBolus(:) = 0.0_RKIND
+            invAreaCell1 = invAreaCell(iCell)
+            do i = 1, nEdgesOnCell(iCell)
+               iEdge = edgesOnCell(i, iCell)
+               edgeSignOnCell_temp = edgeSignOnCell(i, iCell)
+               dcEdge_temp = dcEdge(iEdge)
+               dvEdge_temp = dvEdge(iEdge)
+               do k = minLevelCell(iCell), maxLevelCell(iCell)
+                  ! Compute vertical velocity from the horizontal GM Bolus velocity
```

Comment:
Just updating the comment to avoid confusion
```suggestion
                  ! Compute vertical velocity from the horizontal MLE Bolus velocity
```

Reply:
Fixed